daemon/routes/netlink/netlink.go

```diff
@@ -0,0 +1,141 @@
+/*
+Package netlink provides router implementation that uses netlink.
+*/
+package netlink
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/daemon/routes"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sys/unix"
+)
+
+// Router uses netlink under the hood.
+type Router struct {
+	routes []routes.Route
+	sync.Mutex
+}
+
+// Add appends route to a routes list via netlink if it does not exist yet.
+func (r *Router) Add(route routes.Route) error {
+	r.Lock()
+	defer r.Unlock()
+	if r.has(route) {
+		return fmt.Errorf("route %+v already exists", route)
+	}
+	// check if such route does not exist in routing table
+	netlinkRoute := toNetlinkRoute(route)
+	exists, err := routeExists(toNetlinkRoute(route))
+	if err != nil && exists {
+		return fmt.Errorf("checking if route %+v exists: %w", route, err)
+	}
+	// If route already existed in the system, do not add it to the router so it would not be
+	// flushed.
+	if exists {
+		return nil
+	}
+
+	err = netlink.RouteAdd(&netlinkRoute)
+	if err != nil {
+		if errors.Is(err, os.ErrExist) {
+			return routes.ErrRouteToOtherDestinationExists
+		}
+		return fmt.Errorf("adding route %+v to a routing table: %w", route, err)
+	}
+
+	r.routes = append(r.routes, route)
+	return nil
+}
+
+// Flush deletess all previously added routes via netlink.
+func (r *Router) Flush() error {
+	r.Lock()
+	defer r.Unlock()
+	for _, route := range r.routes {
+		netlinkRoute := toNetlinkRoute(route)
+		exists, err := routeExists(netlinkRoute)
+		if err != nil {
+			return fmt.Errorf("retrieving route %+v: %w", route, err)
+		}
+		if !exists {
+			return fmt.Errorf("route %+v does not exist", route)
+		}
+		if err := netlink.RouteDel(&netlinkRoute); err != nil {
+			return fmt.Errorf("deleting route %+v: %w", route, err)
```

Comment:
In case of errors in cases such as this(cleanup in a loop), perhaps it would be better to log an error but attempt continue with the cleanup? It might be possible to remove other rules.

Reply:
In most cases flush error is already just printed. So I did the following:
1. Cleanup is resumed after error, all errors are gathered and joined with `errors.Join`
2. Changed the remaining places of flush usage to log the warning